File: app/JsDoc/js/src/Conf.js
// Copyright 24-Sep-2017 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

goog.provide("Conf");

Conf = class {
  /**
   * @param {string} path
   * @param {string} lang
   * @param {boolean} showAll
   */
  constructor (path, lang, showAll) {
    /** @private */
    this._path = path;
    /** @private */
    this._lang = lang;
    /** @private */
    this._showAll = showAll;
  }

  /** @return {string} */
  path () {
    return this._path;
  }

  /** @param {string} value */
  setPath (value) {
    this._path = value;
  }

  /** @return {string} */
  lang () {
    return this._lang;
  }

  /** @param {string} value */
  setLang (value) {
    this._lang = value;
  }

  /** @return {boolean} */
  showAll () {
    return this._showAll;
  }

  /** @param {boolean} value */
  setShowAll (value) {
    this._showAll = value;
  }

  /** @return {!Array<?>} */
  serialize () {
    return [
      this._path,
      this._lang,
      this._showAll
    ];
  }

  /**
   * @param {!Array<?>} serial
   * @return {!Conf}
   */
  static restore (serial) {
    return new Conf (
      serial[0],
      serial[1],
      serial[2]
    );
  }
}

File: app/MBackup/go/db/conf/conf.go
// Copyright 05-Sep-2020 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

// Settings data base.
package conf

import (
	"github.com/dedeme/golib/file"
	"github.com/dedeme/golib/json"
	"path"
)

const (
	langKey = "lang"
)

var fpath string

// Auxiliar function
func write(cf map[string]json.T) {
	file.WriteAll(fpath, json.Wo(cf).String())
}

// Auxiliar function
func read() map[string]json.T {
	return json.FromString(file.ReadAll(fpath)).Ro()
}

// Auxiliar function
func getField(key string) json.T {
	return read()[key]
}

// Auxiliar function
func setField(key string, js json.T) {
	cf := read()
	cf[key] = js
	write(cf)
}

// Intializes table.
//    parent: Parent directory of "Conf.tb".
func Initialize(parent string) {
	fpath = path.Join(parent, "Conf.tb")
	if !file.Exists(fpath) {
		write(map[string]json.T{
			langKey: json.Ws("es"),
		})
	}
}

// Gets web language.
func Lang() string {
	return getField(langKey).Rs()
}

// Sets web language.
//    lang: 'es' or 'en'.
func SetLang(lang string) {
	setField(langKey, json.Ws(lang))
}
